scipy/stats/tests/test_variation.py

```diff
@@ -1,159 +1,197 @@
+    def test_propagate_nan(self, xp):
         # Check that the shape of the result is the same for inputs
         # with and without nans, cf gh-5817
-        a = np.arange(8).reshape(2, -1).astype(float)
-        a[1, 0] = np.nan
+        a = xp.reshape(xp.arange(8, dtype=float), (2, -1))
+        a[1, 0] = xp.nan
         v = variation(a, axis=1, nan_policy="propagate")
-        assert_allclose(v, [np.sqrt(5/4)/1.5, np.nan], atol=1e-15)
+        xp_assert_close(v, [math.sqrt(5/4)/1.5, xp.nan], atol=1e-15)
 
-    def test_axis_none(self):
+    @skip_xp_backends(np_only=True, reasons=['Python list input uses NumPy backend'])
+    def test_axis_none(self, xp):
         # Check that `variation` computes the result on the flattened
         # input when axis is None.
         y = variation([[0, 1], [2, 3]], axis=None)
-        assert_allclose(y, np.sqrt(5/4)/1.5)
+        xp_assert_close(y, math.sqrt(5/4)/1.5)
 
-    def test_bad_axis(self):
+    def test_bad_axis(self, xp):
         # Check that an invalid axis raises np.exceptions.AxisError.
-        x = np.array([[1, 2, 3], [4, 5, 6]])
-        with pytest.raises(AxisError):
+        x = xp.asarray([[1, 2, 3], [4, 5, 6]])
+        with pytest.raises((AxisError, IndexError)):
             variation(x, axis=10)
 
-    def test_mean_zero(self):
+    def test_mean_zero(self, xp):
         # Check that `variation` returns inf for a sequence that is not
         # identically zero but whose mean is zero.
-        x = np.array([10, -3, 1, -4, -4])
+        x = xp.asarray([10., -3., 1., -4., -4.])
         y = variation(x)
-        assert_equal(y, np.inf)
+        xp_assert_equal(y, xp.asarray(xp.inf))
 
-        x2 = np.array([x, -10*x])
+        x2 = xp.stack([x, -10.*x])
         y2 = variation(x2, axis=1)
-        assert_equal(y2, [np.inf, np.inf])
+        xp_assert_equal(y2, xp.asarray([xp.inf, xp.inf]))
 
     @pytest.mark.parametrize('x', [np.zeros(5), [], [1, 2, np.inf, 9]])
-    def test_return_nan(self, x):
+    def test_return_nan(self, x, xp):
+        x = xp.asarray(x)
         # Test some cases where `variation` returns nan.
-        y = variation(x)
-        assert_equal(y, np.nan)
+        with suppress_warnings() as sup:
+            # torch
+            sup.filter(UserWarning, "std*")
+            y = variation(x)
+        xp_assert_equal(y, xp.asarray(xp.nan, dtype=x.dtype))
 
     @pytest.mark.parametrize('axis, expected',
                              [(0, []), (1, [np.nan]*3), (None, np.nan)])
-    def test_2d_size_zero_with_axis(self, axis, expected):
-        x = np.empty((3, 0))
-        y = variation(x, axis=axis)
-        assert_equal(y, expected)
-
-    def test_neg_inf(self):
+    def test_2d_size_zero_with_axis(self, axis, expected, xp):
+        x = xp.empty((3, 0))
+        with suppress_warnings() as sup:
+            # torch
+            sup.filter(UserWarning, "std*")
+            y = variation(x, axis=axis)
+        xp_assert_equal(y, xp.asarray(expected))
```

Comment:
@mdhaber not sure how to get this test to pass because of how `axis=None` is handled in `@_axis_nan_policy_factory`

Reply:
You mean that for NumPy, `axis=None` *is* handled in the decorator, but not for other backends? Fine with me to add that bit of code that ravels the input if `axis is None` at the top.